src/components/alert/icon.converter.ts

```diff
@@ -0,0 +1,11 @@
+import type { ComplexAttributeConverter } from 'lit';
+
+export const iconConverter = (): ComplexAttributeConverter<string | boolean> => {
+  return {
+    fromAttribute: (value: string): string | boolean => {
+      if (!value || value === 'true') return true;
```

Comment:
Issue here that we use `icon` attribute as `boolean` and also `string` to change default icon. Lit's default converter does not work with multiple converters, so we needed to use this and this converter requires single type and if we use `boolean` as value type we can not catch when usage is like `icon="download"`. For this case when we use attribute as `icon` this value does not work like when attribute type is `boolean`. 
These cases requires to use converter hence requires a type check when using `getIcon()` function.

Reply:
Ok, I got your point. I also just checked [Lit document about custom converters](https://lit.dev/docs/components/properties/#conversion-converter). So you mean, if we don't use this converter and user sets `<bl-alert icon>` then value of the `icon` attribute becomes a string `true`, right? Did you check `value` here can ever be string `true` or `false`?

If this is a generic way of having an attribute with `boolean | string` type, maybe we can move this to our utlities folder with a more generic name to make it reusable in other components.

Reply:
> Ok, I got your point. I also just checked Lit document about custom converters. So you mean, if we don't use this converter and user sets <bl-alert icon> then value of the icon attribute becomes a string true, right? Did you check value here can ever be string true or false?

for this ( `<bl-alert icon>` ) kind of usage It actually becomes empty string, instead of string `true`, if attribute type (or converter value type) is set to `boolean` then it converts to boolean `true`. 
In converter that's why i've added `!value` because if user adds `icon` attribute on component it should work as `boolean` `true` and `!value` covers this section.

> If this is a generic way of having an attribute with boolean | string type, maybe we can move this to our utlities folder with a more generic name to make it reusable in other components.

I agree to that, i think it's highly possible that we'll face another scenario like this. But i think we might need `date` to `string` converters like Lit tutorials mension. So instead of `utilities` folder, we might consider to have another folder called `converters`. What do you think?

Reply:
addition to that, if we've agreed on i'm signing this comment https://github.com/Trendyol/baklava/pull/217/files#r963506474 to resolved.

Reply:
As far as I understand, value in this function never can be string `true` or string `false`. If so, we don't need to run a logic over this potential.

We can re-organize folder structure in the future if we'll have more and more converters. For now, I think just utilities folder is good enough for making this a common helper that multiple components can use.

Reply:
> addition to that, if we've agreed on i'm signing this comment https://github.com/Trendyol/baklava/pull/217/files#r963506474 to resolved.

I don't think this is directly related with here. Let me write some more explanation there.